src/Core/Managed/Shared/Extensibility/TelemetrySink.cs

```diff
@@ -0,0 +1,149 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Represents a destination for telemetry, consisting of a set of telemetry processors and a channel.
+    /// </summary>
+    public sealed class TelemetrySink : IDisposable, ITelemetryProcessor
+    {
+        private TelemetryConfiguration telemetryConfiguration;
+        private ITelemetryChannel telemetryChannel;
+        private bool shouldDisposeChannel;
+        private TelemetryProcessorChain telemetryProcessorChain;
+        private TelemetryProcessorChainBuilder telemetryProcessorChainBuilder;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelemetrySink"/> class.
+        /// </summary>
+        /// <param name="telemetryConfiguration">Telemetry configuration to use for the new <see cref="TelemetrySink"/> instance.</param>
+        /// <param name="telemetryChannel">Telemetry channel to use for the new <see cref="TelemetrySink"/> instance.</param>
+        public TelemetrySink(TelemetryConfiguration telemetryConfiguration, ITelemetryChannel telemetryChannel = null)
```

Comment:
I mostly made TelemetrySink an ITelemetryProcessor out of convenience (can treat all the elements in the pipeline uniformly).

If you feel strongly about the requirement that ITelemetryProcessor has to have the abovementioned constructor, I'd rather remove ITelemetryProcessor designation from TelemetrySink. The constructor taking "next" processor just does not make sense for TelemetrySink.